pkg/controller/job/job_controller.go

```diff
@@ -421,6 +423,33 @@ func (jm *Controller) updateJob(old, cur interface{}) {
 	}
 }
 
+// deleteJob enqueues the job and all the pods associated with it that still
+// have a finalizer.
+func (jm *Controller) deleteJob(obj interface{}) {
+	jm.enqueueController(obj, true)
+	jobObj, ok := obj.(*batch.Job)
+	if !ok {
+		tombstone, ok := obj.(cache.DeletedFinalStateUnknown)
+		if !ok {
+			utilruntime.HandleError(fmt.Errorf("couldn't get object from tombstone %+v", obj))
+			return
+		}
+		jobObj, ok = tombstone.Obj.(*batch.Job)
+		if !ok {
+			utilruntime.HandleError(fmt.Errorf("tombstone contained object that is not a job %+v", obj))
+			return
+		}
+	}
+	// Listing pods shouldn't really fail, as we are just querying the informer cache.
+	pods, _ := jm.podStore.Pods(jobObj.Namespace).List(labels.Everything())
```

Comment:
Why not using the job selector when listing pods? We don't need to be looking at all pods, especially that it already happens in `syncJob` when adopting pods.

Reply:
Done